storage/src/main/java/tech/pegasys/teku/storage/store/Store.java

```diff
@@ -858,13 +858,10 @@ private SafeFuture<Optional<StateAndBlockSummary>> getOrRegenerateBlockAndState(
       return SafeFuture.completedFuture(maybeEpochState);
     }
 
-    // if finalized is gone from cache we can still reconstruct that without regenerating
+    // if finalized is gone from cache we can use the finalized anchor without regenerating
     if (finalizedAnchor.getRoot().equals(blockRoot)) {
       LOG.trace("epochCache GET finalizedAnchor {}", finalizedAnchor::getSlot);
-      return SafeFuture.completedFuture(
-          Optional.of(
-              StateAndBlockSummary.create(
-                  finalizedAnchor.getBlockSummary(), finalizedAnchor.getState())));
+      return SafeFuture.completedFuture(Optional.of(finalizedAnchor));
```

Comment:
The stateRoot of the anchor is still the block state root though

Reply:
AnchorPoint.getStateRoot() still returns the state_root of the block, not the state

Reply:
Method ` private SafeFuture<Optional<StateAndBlockSummary>> getOrRegenerateBlockAndState(
final Bytes32 blockRoot)` should return state at block with blockRoot, transitioned state is not this state. This method is used by a lot of different consumers, it could be an issue

Reply:
Yeah `AnchorPoint.getStateRoot()` returns the `state_root` of the `blockSummary`, so would be the state associated with the block

Reply:
In any case, `StateAndBlockSummary.create(finalizedAnchor.getBlockSummary(), finalizedAnchor.getState());` is same as just returning `finalizedAnchor`